### PR TITLE
[Compatibility] Improve python3 detection on windows

### PR DESCRIPTION
**Pull request by roadrunner1776**
Bug fix: don't fail on (Windows) systems without python installed

### CODE REVIEW

File: src/elisp/treemacs-customization.el [hunk 1]
```diff
@@ -32,11 +32,12 @@
            (s-trim)
            (s-lines)
            (--first
-            (->> (concat (shell-quote-argument it) " --version")
+            (if (file-exists-p it)
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-customization.el]: Without an else branch this should be a `ẁhen`.

Comment (roadrunner1776) [reply src/elisp/treemacs-customization.el]: Replaced the if statement with a when statement

**Comment by Alexander-Miller** [comment]
:+1:

**Comment by duianto** [comment]
This current version returns `nil` for me:
https://github.com/Alexander-Miller/treemacs/blob/ba34423a12bcdbf74e0afa4bd4bd5ff877251c08/src/elisp/treemacs-customization.el#L26

The previous version returns the path to python 3.
`"c:\\Python37\\python.exe"`

https://github.com/Alexander-Miller/treemacs/blob/2b652d8273de012bc3a825cfc6cc73f4c03906da/src/elisp/treemacs-customization.el#L25

**Pull request merged**